src/Iteration/NondecreasingIndices.jl

```diff
@@ -0,0 +1,85 @@
+"""
+    NondecreasingIndices
+
+Iterator over the vectors of `m` nondecreasing indices from 1 to `n`.
+
+### Fields
+
+- `n` -- size of the index domain
+- `m` -- number of indices to choose (resp. length of the vectors)
+
+### Notes
+
+The vectors are modified in-place.
+
+The iterator ranges over ``\\binom{n + m - 1}{m}`` (`n + m - 1` choose `m`)
+possible vectors.
+
+This implementation results in a lexicographic order with the last index growing
+first.
+
+See also [`StrictlyIncreasingIndices`](@ref) for a strictly increasing iterator.
+
+### Examples
+
+```jldoctest; setup = :(using ReachabilityBase.Iteration)
+julia> for v in NondecreasingIndices(4, 2)
+           println(v)
+       end
+[1, 1]
+[1, 2]
+[1, 3]
+[1, 4]
+[2, 2]
+[2, 3]
+[2, 4]
+[3, 3]
+[3, 4]
+[4, 4]
+```
+"""
+struct NondecreasingIndices
+    n::Int
+    m::Int
+
+    function NondecreasingIndices(n::Int, m::Int)
+        new(n, m)
+    end
```

Comment:
This is fine, but note that it is not required as it is the same as the default constructor i think.

Reply:
True, I copy-pasted this from `StrictlyIncreasingIndices`, but here we do not need the assertion :+1: